docs/send-tax-receipts-for-donations/README.md

```diff
@@ -0,0 +1,54 @@
+
+* View in the task library: [tasks.mechanic.dev/send-tax-receipts-for-donations](https://tasks.mechanic.dev/send-tax-receipts-for-donations)
+* Task JSON, for direct import: [task.json](../../tasks/send-tax-receipts-for-donations.json)
+* Preview task code: [script.liquid](./script.liquid)
+
+## Default options
+
+```json
+{
+  "tax_id__required": null,
+  "only_include_orders_with_any_of_these_tags__array": null,
+  "send_cc_email_to_shop__boolean": false,
+  "email_subject__required": "Receipt {{ order.name }} for donation to {{ shop.name }}",
+  "email_body__multiline_required": "Thank you for your recent donation!\n\nAttached you will find your official donation receipt.",
+  "donation_receipt_filename__required": "{{ shop.name | handleize }}-donation-receipt-{{ order.name  }}.pdf",
+  "donation_reciept_html_template__code_multiline_required": "<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n  <meta charset=\"utf-8\">\n  <style>\n    .container {\n      width: 960px;\n      margin: 0 auto;\n      font-size: 16px;\n    }\n    .header {\n      display: flex;\n      flex-direction: row;\n      border-top: 1px solid #000; \n      border-bottom: 1px dashed #000; \n      font-size: 18px;\n    }\n    .main {\n      margin-top: 10px;\n      display: flex;\n      flex-direction: row;\n      font-size: 16px;\n    }\n    .left {\n      width: 50%;\n    }\n    .right {\n      width: 50%;\n    }\n  </style>\n</head>\n<body>\n  <div class=\"container\">\n    <div class=\"header\">\n      <p class=\"left\">\n        {{ shop.name }}<br/>\n        {{ shop.address1 }}<br/>{% if shop.address2 != blank -%}{{ shop.address2 }}<br/>{%- endif %}\n        {{ shop.city }}, {{ shop.province }} {{ shop.zip }}<br/>\n        {{ order.billing_address.country }}\n      </p>\n      <p class=\"right\">\n        OFFICIAL DONATION RECEIPT<br/>\n        TAX ID #[TAX_ID]<br/>\n      </p>\n    </div>\n    <div class=\"main\">\n      <p class=\"left\">\n        <strong>Donor:</strong><br/>\n        {{ order.billing_address.first_name }} {{ billing_address.last_name }}<br/>\n        {{ order.billing_address.address1 }}<br/>{% if order.billing_address.address2 != blank -%}{{ order.billing_address.address2 }}<br/>{%- endif %}\n        {{ order.billing_address.city }}, {{ order.billing_address.province }} {{ order.billing_address.zip }}<br/>\n        {{ order.billing_address.country }}\n      </p>\n      <p class=\"right\">\n        <strong>Donation Details:</strong><br/>\n        Order Number: {{ order.name }}<br/>\n        Donation Received: {{ order.created_at | date: \"%F\" }}<br/>\n        Amount: {{ order.total_price | currency }}<br/>\n        Date Issued: {{ \"now\" | date: \"%F\" }}<br/>\n        Place Issued: {{ shop.province }}, {{ shop.country_name }}<br/>\n        <br/>\n        Donations are tax deductible to the extent permitted by law.<br/>\n        <br/>\n        Thank you\n      </p>\n    </div>\n  </div>\n</body>\n</html>"
```

Comment:
@kevinhughes27  That's a good point. In your experience, do merchants typically allow donation items to be purchased alongside other products, or do they isolate the donation purchases as much as possible? I kind of assumed the latter, which is why the task checks for a donation order tag.

If mixing donations with non-donation products is fairly common, then instead of an order tag, the task would need to check for identifying information on the line item as you say. The catch here is that there are several ways this could be implemented - product / variant titles; product types; line item attributes. It really wouldn't be practical to allow for all of those methods, so we should choose the one that is likely to be used in most scenarios.

Reply:
I think they do often have orders that include regular items and donations just based on the way my app was configured. If a shop has real products for purchase and donation products it would be tough (maybe impossible?) for them to control keeping them separate.

Could we use a product tag to control this? That would seem the most logical to me. Then a merchant would tag all their donation products as such and the task would just work. My app was configured at the product level and that has been working fine for a long time. This seems simpler than doing anything at the variant level. For any more advanced cases we can always refer the merchant to edit their task 😄

Reply:
Alrighty, we'll go with product tags. Thank you for the feedback.